runtime/stdlib/test.go

```diff
@@ -1472,12 +1483,12 @@ func newMatcherWithGenericTestFunction(
 
 	inter := invocation.Interpreter
 
-	staticType, ok := testFunc.StaticType(inter).(interpreter.FunctionStaticType)
-	if !ok {
+	typeParameterPair := invocation.TypeParameterTypes.Oldest()
+	if typeParameterPair == nil {
 		panic(errors.NewUnreachableError())
 	}
 
-	parameters := staticType.Type.Parameters
+	parameterType := typeParameterPair.Value
```

Comment:
Fix: Use the type argument for type parameter `T` of `Test.newMatcher<T>`: The wrapper should check if the run-time type of argument passed to the test function (`Matcher.test` has type  `((AnyStruct): Bool)`) matches the type argument `T`